modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java

```diff
@@ -0,0 +1,251 @@
+        locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        EntityUtils.consume(response.getEntity());
+        response = sendGetRequest(httpClientWithoutAutoRedirections, locationHeader.getValue());
+        HttpClientBuilder.create().setDefaultCookieStore(cookieStore);
+        Map<String, Integer> keyPositionMap = new HashMap<>(1);
+        keyPositionMap.put("name=\"sessionDataKeyConsent\"", 1);
+        List<DataExtractUtil.KeyValue> keyValues = DataExtractUtil.extractSessionConsentDataFromResponse(response,
+                keyPositionMap);
+        Assert.assertNotNull(keyValues, "SessionDataKeyConsent key value is null for " + application
+                .getApplicationName());
+
+        sessionDataKeyConsent = keyValues.get(0).getValue();
+        Assert.assertNotNull(sessionDataKeyConsent, "Invalid sessionDataKeyConsent for " + application
+                .getApplicationName());
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private static String claimsToGetConsent(HttpResponse response) throws Exception {
+
+        String redirectUrl = Utils.getRedirectUrl(response);
+        Map<String, String> queryParams = Utils.getQueryParams(redirectUrl);
+        String requestedClaims = queryParams.get("requestedClaims");
+        String mandatoryClaims = queryParams.get("mandatoryClaims");
+        if (StringUtils.isNotEmpty(requestedClaims)) {
+            return requestedClaims;
+        }
+        if (StringUtils.isNotEmpty(mandatoryClaims)) {
+            return mandatoryClaims;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private static String getScopes(HttpResponse response) throws Exception {
+        String redirectUrl = Utils.getRedirectUrl(response);
+        Map<String, String> queryParams = Utils.getQueryParams(redirectUrl);
+        Assert.assertTrue(queryParams.containsKey("scope"));
+        return queryParams.get("scope");
+    }
+
+
+    private static List<NameValuePair> getNameValuePairs(OIDCApplication application) {
+
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("grantType", OAuth2Constant.OAUTH2_GRANT_TYPE_CODE));
+        urlParameters.add(new BasicNameValuePair("consumerKey", application.getClientId()));
+        urlParameters.add(new BasicNameValuePair("callbackurl", application.getCallBackURL()));
+        urlParameters.add(new BasicNameValuePair("authorizeEndpoint", OAuth2Constant.APPROVAL_URL));
+        urlParameters.add(new BasicNameValuePair("authorize", OAuth2Constant.AUTHORIZE_PARAM));
+        urlParameters.add(new BasicNameValuePair("scope", OAuth2Constant.OAUTH2_SCOPE_OPENID + " " +
+                OAuth2Constant.OAUTH2_SCOPE_EMAIL + " " + OAuth2Constant.OAUTH2_SCOPE_PROFILE + " " +
+                INTERNAL_LOGIN_SCOPE + " " + INTERNAL_USER_MGT_UPDATE_SCOPE + " " + INVALID_SCOPE + " " +
+                SYSTEM_SCOPE));
```

Comment:
Isn't there's a util method that does this? Let's try to reuse it and modify the "scope" param

Reply:
There's util method available to do this. But that method only adds `openid`, `email`, and `profile` scopes. Here I need to add some additional scopes as well, That's why I extracted that code snippet from the util class. I cannot add additional scopes after the object has been added to the list.